src/azip/azure.go

```diff
@@ -119,14 +120,15 @@ func getNIC(nicClient network.InterfacesClient, vm compute.VirtualMachine, group
 			if err != nil {
 				return nil, err
 			}
-			if (nic.InterfacePropertiesFormat == nil) || ((*nic.InterfacePropertiesFormat).Primary == nil) {
+			if nic.Tags == nil {
 				continue
```

Comment:
Can we retain the check for the Primary field here (i.e. when not Tags is specified) to maintain compatibility with existing behavior? The priority for selecting then will be (1) Look for tag and (2) select Primary if no Tag specified. You can also add a log to state on what basis the NIC is chosen.

Reply:
Done